packages/gensx-vercel-ai/src/tools.ts

```diff
@@ -0,0 +1,29 @@
+import type { InferToolParams, ToolBox } from "@gensx/core";
+
+import { executeExternalTool } from "@gensx/core";
+import { jsonSchema, type Tool } from "ai";
+import z4 from "zod/v4";
+
+// Utility function to convert ToolBox to Vercel AI SDK ToolSet format
+export function asToolSet(toolBox: ToolBox): Record<string, Tool> {
+  return Object.entries(toolBox).reduce<Record<string, Tool>>(
+    (acc, [name, toolDef]) => {
+      let paramsSchema;
+      if ("_zod" in toolDef.params) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-explicit-any
+        paramsSchema = jsonSchema(z4.toJSONSchema(toolDef.params) as any);
+      } else {
+        paramsSchema = toolDef.params;
+      }
+      acc[name] = {
+        description: toolDef.description ?? `Tool: ${name}`,
```

Comment:
nit: description is optional ([docs](https://ai-sdk.dev/docs/reference/ai-sdk-core/generate-text#tools.tool.description)) so it's probably best not to muck with it 
```suggestion
        description: toolDef.description,
```

Reply:
ahh i see, you're doing this because it's optional in the AI SDK but not on our end. I think we should probably just make it optional on our end to align with what the ai sdk does.